File: RestaurantServer/src/main/java/dk/restaurant/services/impl/RestaurantService.java
package dk.restaurant.services.impl;

import dk.restaurant.models.Restaurant;
import dk.restaurant.network.IClientFactory;
import dk.restaurant.network.IRestaurantClient;
import dk.restaurant.services.IRestaurantService;

public class RestaurantService implements IRestaurantService {

    private IRestaurantClient client;

    public RestaurantService(IClientFactory clientFactory) {
        this.client = clientFactory.getRestaurantClient();
    }

    @Override
    public Restaurant getRestaurant() {
        System.out.println("here service");
        return client.getRestaurant() ;
    }
}

File: WebClient/Data/Impl/RestaurantService.cs
﻿using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using WebClient.Data.validationhandler;
using WebClient.Models;

namespace WebClient.Data.Impl
{
    public class RestaurantService:IRestaurantService
    {
        private readonly HttpClient client;
        private JsonSerializerOptions options;
        private string uri = "http://localhost:8080";

        public RestaurantService()
        {
            client = new HttpClient();
            options = new JsonSerializerOptions()
            {
                PropertyNamingPolicy = JsonNamingPolicy.CamelCase
            };
        }
        public async Task<Restaurant> GetRestaurant()
        {
            HttpResponseMessage response = await client.GetAsync($"{uri}/restaurant");
       
            if 
                (!response.IsSuccessStatusCode)
            {
                var errorMessage = await response.Content.ReadAsStringAsync();
                var exceptionResponse = JsonSerializer.Deserialize<ExceptionResponse>(errorMessage, options); 
                throw new Exception($"{exceptionResponse.Message}");
            }
            string result = await response.Content.ReadAsStringAsync();
            Restaurant restaurant = JsonSerializer.Deserialize<Restaurant>(result, options);
            return restaurant;        }
    }
}